parquet-hadoop/src/main/java/org/apache/parquet/hadoop/rewrite/ParquetRewriter.java

```diff
@@ -183,12 +186,61 @@ public ParquetRewriter(TransParquetFileReader reader,
     }
   }
 
+  // Open all input files to validate their schemas are compatible to merge
+  private void openInputFiles(List<Path> inputFiles, Configuration conf) {
+    Preconditions.checkArgument(inputFiles != null && !inputFiles.isEmpty(), "No input files");
+
+    for (Path inputFile : inputFiles) {
```

Comment:
Do we want to set a max size?

Reply:
It is tricky to set a max size, whether in terms of number of files or total file sizes. Based on my knowledge, the performance varies when input files have different number of rows, number of columns, number of blocks, block sizes, etc. So it would be better for the caller to do the planning.